### ISSUE TITLE
marko-body="static-text" attribute fails when HTML node of same type is within escaped block

### ISSUE THREAD

**Issue by jskorupski** [marko-js/marko#343]
### Context

I have a project where we embed another templating language inside of a marko template. Because of this, there are cases where we need to disable Marko's parsing for sections of HTML, so that Marko does not try to parse foreign template syntax that may throw errors. We use the marko-body="static-text" attribute to disable Marko parsing. 

HTML snippets that causes parser error:

Example 1:
`<div marko-body="static-text">
    <div>Content Goes Here</div>
</div>`

Example 2:
`<section marko-body="static-text">
        <div>Other Content</div>
    <section>Content Goes Here</section>
</section>`

Note that the element type of node with marko-body attribute must match some element within it. 
### Expected Behavior

The output should be:

Example 1:
`<div>
    <div>Content Goes Here</div>
</div>`

Example 2:
`<section>
        <div>Other Content</div>
    <section>Content Goes Here</section>
</section>`
### Actual Behavior

Example 1: Parser error - 'The closing "div" tag was not expected'
Example 1: Parser error - 'The closing "section " tag was not expected'
### Possible Fix

<details><summary>

Additional Info</summary>

### Your Environment
- Versions used:
- Marko 3.8.1
- Lasso 2.5.5
- Lasso-Marko 2.0.7
- Operating System and version (desktop or mobile): Ubuntu 14.04
- Link to your project: (eBay Internal)
### Steps to Reproduce

Input either of these pieces of code on http://markojs.com/try-online/

`<div marko-body="static-text">
    <div>Content Goes Here</div>
</div>`

`<section marko-body="static-text">
        <div>Other Content</div>
    <section>Content Goes Here</section>
</section>`
### Stack Trace

Example 1: Parser error - 'The closing "div" tag was not expected'
Example 1: Parser error - 'The closing "section " tag was not expected'

</details>

**Comment by patrick-steele-idem** [comment]
Some notes:

- The parser is working as expected
- This is not really a bug, but it is a "gotcha"
- We should offer a custom tag to avoid the situation where nested body content is ended too early.

One possible solution:

```xml
<marko-parser body='static-text'>
    <section>
        <div>Other Content</div>
        <section>Content Goes Here</section>
    </section>
</marko-parser>
```

Thoughts?

**Comment by jskorupski** [comment]
A custom tag would be great! Another huge benefit of a custom marko-parser tag (versus attribute on a rendered tag) is that it can act a non-rendered parser directive, resulting in no additional HTML output/DOM structure change. 

Background: When I used the existing version of this marko-body attribute, I wanted to be able to ignore a block of HTML (which contained non-marko compatible AngularJS syntax in it), but not result in a new non-functional node that wraps the content (which can potentially affect animation and browser compositing performance depending on the kind of operations you perform on your DOM elements). I ended up attaching the marko-body attribute to each HTML element, but had to be careful of the same element type within to prevent this gotcha.
